File: binarysearch.io/258_lowest_common_ancestor.cpp
/**
 * class Tree {
 *     public:
 *         int val;
 *         Tree *left;
 *         Tree *right;
 * };
 */
int solve(Tree* root, int a, int b) {
    if(root==NULL) return -1;
    int left = solve(root->left,a,b);
    int right = solve(root->right,a,b);
    if(left!=-1 && right!=-1) return root->val;
    if(root->val==a || root->val==b) return root->val;
    return left!=-1?left:right;   
}

File: binarysearch.io/258_lowest_common_ancestor.py
# class Tree:
#     def __init__(self, val, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def solve(self, root, a, b):
        if root is None:
            return -1
        left = self.solve(root.left,a,b)
        right = self.solve(root.right,a,b)
        if (root.val == a or root.val == b) and (left!=-1 or right!=-1):
            return root.val 
        
        if left!=-1 and right!=-1:
            return root.val
        
        if root.val==a or root.val==b:
            return root.val 
        
        if left!=-1: return left
        else: return right 
            
        
        
        
        
        
        